framework/src/userobjects/PropertyReadFile.C

```diff
@@ -50,14 +53,23 @@ PropertyReadFile::validParams()
       "Periodic or non-periodic grain distribution: Default is non-periodic");
   params.addParam<bool>(
       "use_zero_based_block_indexing", true, "Are the blocks numbered starting at zero?");
+
+  // Set an execution schedules to what makes sense currently
+  // We do not allow INITIAL because we read the file at construction
+  ExecFlagEnum & exec_enum = params.set<ExecFlagEnum>("execute_on", true);
+  exec_enum.removeAvailableFlags(EXEC_INITIAL, EXEC_FINAL, EXEC_LINEAR, EXEC_NONLINEAR);
+  params.setDocString("execute_on", exec_enum.getDocString());
+
   return params;
 }
 
 PropertyReadFile::PropertyReadFile(const InputParameters & parameters)
   : GeneralUserObject(parameters),
-    _prop_file_name(getParam<FileName>("prop_file_name")),
-    _reader(_prop_file_name),
-
+    _prop_file_names(getParam<std::vector<FileName>>("prop_file_name")),
+    _current_file_index(declareRestartableData<unsigned int>(name() + "_file_index", 0)),
```

Comment:
I assume you're talking about this. Yea I mean this looks fine to me but I'm not the restart expert

Reply:
yup this one. I tried int it was the same problem. Is the name() the bad idea? I ll try that